### ISSUE TITLE
Persist data on our Backend IO Apps (graphql or node API)

### ISSUE THREAD

**Issue by guilhermebruzzi** [vtex/typescript#19]
**What is to be discussed?**

For GET request, should we only use the HTTP Cache-control headers and trust the browser?

For caching values, should we use Master Data, VBase that comes with IO, or In-memory DataStores like our Redis clusters?

For persisting values that we would like to retrieve after should we use Master Data, VBase that comes with IO or our Redis clusters?

**Describe the solution you'd like**

It depends on each case.

If to assemble the response is fast enough, don't even persist or cache anything.

HTTP Cache-control is good enough for resources that don't change based on the url (not suited for graphql) and which is fast enough to assemble the response.

Redis is more suitable for caching instead of persisting values and is faster than persist on VBase, but I've never used on a IO App because it wasn't possible when I wanted (but if it makes sense to be a pattern, we should demand it)

VBase is good enough when you want to store data based on key / value

Master Data is more suitable for complex data with relations that you want to do a search and bring related data

**Comment by luciannojunior** [comment]
@saviofelixmuniz has solid plans on doing a default MasterData client to be exported from `@vtex/api`, what would help a lot. He also have been talking about plans to create a UI admin to MasterData, what I think that, when done, could put MD closer to almost all solutions

**Comment by guilhermebruzzi** [comment]
@luciannojunior Redis is still better than MD for caching key-value based values. But for all other cases this would make MD way better than VBase for persisting data

**Comment by marcosvcp** [comment]
This topic is really related to js-standard?

**Comment by guilhermebruzzi** [comment]
@marcosvcp No, but I think the original idea were also about patterns that the frontend teams would follow on their products, including on the graphql stack, right @kaisermann ?
